qiskit/chemistry/algorithms/ground_state_solvers/minimum_eigensolver_factories/vqe_uvccsd_factory.py

```diff
@@ -0,0 +1,145 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The minimum eigensolver factory for ground state calculation algorithms."""
+
+from typing import Optional
+import numpy as np
+
+from qiskit.aqua import QuantumInstance
+from qiskit.aqua.algorithms import MinimumEigensolver, VQE
+from qiskit.aqua.operators import ExpectationBase
+from qiskit.aqua.components.optimizers import Optimizer
+from qiskit.chemistry.components.initial_states import VSCF
+from qiskit.chemistry.components.variational_forms import UVCC
+from qiskit.chemistry.transformations import BosonicTransformation
+
+from .minimum_eigensolver_factory import MinimumEigensolverFactory
+
+
+class VQEUVCCSDFactory(MinimumEigensolverFactory):
+    """A factory to construct a VQE minimum eigensolver with UVCCSD ansatz wavefunction."""
```

Comment:
Why do we call it `VQEUVCCSD` when the var for is just `UVCC` so `VQEUVCC` seems more applicable

Reply:
because in this specific factory it asks for singles and doubles to be included.
the variational form is UVCC in general but UVCCSD if you specifically include the singles and doubles

Reply:
UVCC can possibly be extended to UVCCSDTQ... Better to specify it here. This is resolved.